### PR TITLE
Fixes #3160. TextField doesn't update correctly the CursorPosition on Paste.

### LINKED ISSUE #3160: TextField doesn't update correctly the CursorPosition on Paste.
Running this unit test will fail on the last `CursorPosition` asserting.

```cs
	[Fact, TextFieldTestsAutoInitShutdown]
	public void Copy_Paste_Text_Changing_Updates_Cursor_Position ()
	{
		_textField.TextChanging += _textField_TextChanging;

		void _textField_TextChanging (object sender, TextChangingEventArgs e)
		{
			if (e.NewText.GetRuneCount () > 11) {
				e.NewText = e.NewText [..11];
			}
		}

		Assert.Equal (32, _textField.CursorPosition);
		_textField.SelectAll ();
		_textField.Cut ();
		Assert.Equal ("TAB to jump between text fields.", Application.Driver.Clipboard.GetClipboardData ());
		Assert.Equal (string.Empty, _textField.Text);
		Assert.Equal (0, _textField.CursorPosition);
		_textField.Paste ();
		Assert.Equal ("TAB to jump", _textField.Text);
		Assert.Equal (11, _textField.CursorPosition);

		_textField.TextChanging -= _textField_TextChanging;
	}

```

### PR DESCRIPTION
**Pull request by BDisp**
Fixes #3160 - If the `Text` changing is canceled or supersede the `CursorPosition` is now correctly updated.

## Pull Request checklist:

- [x] I've named my PR in the form of "Fixes #issue. Terse description."
- [x] My code follows the [style guidelines of Terminal.Gui](https://github.com/gui-cs/Terminal.Gui/blob/develop/.editorconfig) - if you use Visual Studio, hit `CTRL-K-D` to automatically reformat your files before committing.
- [x] My code follows the [Terminal.Gui library design guidelines](https://github.com/gui-cs/Terminal.Gui/blob/develop/CONTRIBUTING.md)
- [x] I ran `dotnet test` before commit
- [ ] I have made corresponding changes to the API documentation (using `///` style comments)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any poor grammar or misspellings
- [x] I conducted basic QA to assure all features are working

### CODE REVIEW

**Comment by BDisp** [comment]
This PR really depends of the PR #3165, I couldn't avoid it because this one was having some unit tests failing.

File: Terminal.Gui/Views/DateField.cs [hunk 1]
```diff
@@ -48,8 +41,7 @@ public class DateField : TextField {
 	/// <param name="x">The x coordinate.</param>
 	/// <param name="y">The y coordinate.</param>
 	/// <param name="date">Initial date contents.</param>
-	/// <param name="isShort">If true, shows only two digits for the year.</param>
-	public DateField (int x, int y, DateTime date, bool isShort = false) : base (x, y, isShort ? 10 : 12, "") => SetInitialProperties (date, isShort);
+	public DateField (int x, int y, DateTime date) : base (x, y, 12, "") => SetInitialProperties (date);
```
Comment (tig) [inline Terminal.Gui/Views/DateField.cs:44]: While you're in here, would you mind removing these constructors with parameters? See ##3171.

**Review by tig — CHANGES_REQUESTED** [review]
Wow. Nice cleanup and code-reduction! See my comment about constructors.

File: Terminal.Gui/Views/DateField.cs (hunk 1 above)
Comment (BDisp) [reply Terminal.Gui/Views/DateField.cs]: Done.

**Review by tig — APPROVED** [review]

**Pull request merged**